test/e2e/deprecated_e2e_test.go

```diff
@@ -1,6 +1,137 @@
 package e2e
 
-// TODO
+import (
+	"context"
+	_ "embed"
+	"time"
 
-// v1beta1 CRD in installplan fails
-// v1beta1 RBAC in an installplan fails
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
+
+	"github.com/blang/semver/v4"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed testdata/deprecated/fakeCRD.yaml
+var fakeCRD string
+
+//go:embed testdata/deprecated/fakeRBAC.yaml
+var fakeRBAC string
+
+var _ = Describe("Deprecated APIs", func() {
+	BeforeEach(func() {
+		csv := newCSV("test-csv", testNamespace, "", semver.Version{}, nil, nil, nil)
+		Expect(ctx.Ctx().Client().Create(context.TODO(), &csv)).To(Succeed())
+	})
+	AfterEach(func() {
+		TearDown(testNamespace)
+	})
+
+	When("deprecated objects are created in the installplan", func() {
+		// each entry is an installplan with a deprecated resource
+		type payload struct {
+			name string
+			ip   *operatorsv1alpha1.InstallPlan
+		}
+
+		// TODO should we use embed string or []byte
+		//var crdb bytes.Buffer
+		//_, err := crdb.Read(fakeCRD)
+		//Expect(err).To(BeNil())
+		//
+		//var rbacb bytes.Buffer
+		//_, err = crdb.Read(fakeRBAC)
+		//Expect(err).To(BeNil())
+
+		var tableEntries []table.TableEntry
+		tableEntries = []table.TableEntry{
+			table.Entry("contains a v1beta1 CRD", payload{
```

Comment:
the test name no longer matches what's being tested; i.e. it mentions `v1beta1` while it actually creates something else.